packages/contracts-bedrock/test/vendor/Initializable.t.sol

```diff
@@ -225,242 +235,273 @@ contract Initializer_Test is Bridge_Initializer {
+            command[1] = "-c";
             command[2] = string.concat(
                 Executables.echo,
                 " '",
-                contractAbi,
+                ForgeArtifacts.getAbi(contractName),
                 "'",
                 " | ",
                 Executables.jq,
                 " '.[] | select(.name == \"initialize\" and .type == \"function\")'"
             );
-            bytes memory res = Process.run(command);
 
-            // If the contract has an `initialize()` function, the resulting query will be non-empty.
-            // In this case, increment the number of `Initializable` contracts.
-            if (res.length > 0) {
-                // Count Proxy + Impl
-                numContracts_ += 2;
+            // If the contract does not have an `initialize()` function, skip it.
+            if (Process.run(command).length == 0) {
+                continue;
+            }
+
+            // Check if this contract is in the contracts array.
+            assertTrue(
+                _hasMatchingContract(contractName), string.concat("Missing ", contractName, " from contracts array")
+            );
+
+            // If the contract is proxied, check that the proxy is in the contracts array.
+            // Skip predeployed contracts for now since we don't yet keep track of the
+            // implementations inside of the deploy script. We should add support for this in the
+            // future so that we can properly check that the implementations for predeployed
+            // contracts are initialized too.
+            if (ForgeArtifacts.isProxiedContract(contractName) && !ForgeArtifacts.isPredeployedContract(contractName)) {
+                assertTrue(
+                    _hasMatchingContract(string.concat(contractName, "Proxy")),
+                    string.concat("Missing ", contractName, "Proxy from contracts array")
+                );
             }
         }
 
-        // Then get L2 contracts
-        command[2] = string.concat(
-            Executables.find,
-            " src/L2 -type f -exec basename {} \\;",
-            " | ",
-            Executables.sed,
-            " 's/\\.[^.]*$//'",
-            " | ",
-            Executables.jq,
-            " -R -s 'split(\"\n\")[:-1]'"
-        );
-        string[] memory l2ContractNames = abi.decode(vm.parseJson(string(Process.run(command))), (string[]));
+        // Attempt to re-initialize all contracts within the `contracts` array.
+        for (uint256 i; i < contracts.length; i++) {
+            InitializeableContract memory _contract = contracts[i];
+            string memory name = _getRealContractName(_contract.name);
 
-        for (uint256 i; i < l2ContractNames.length; i++) {
-            string memory contractName = l2ContractNames[i];
-            string memory contractAbi = ForgeArtifacts.getAbi(contractName);
+            // Grab the value of the "initialized" storage slot. Must handle special case for the
```

Comment:
Do we need to verify or fix the `InitializableOZv5.t.sol` tests also?

Reply:
Hmmm well it looks like the `InitializableOZv5.t.sol` tests aren't doing a lot of the things that the original test is doing. We probably do want to make sure it's up to date. I will look at that separately.

Reply:
Opened #11766